src/packet_types.c

```diff
@@ -199,6 +199,40 @@ void telemetry_request_block_init(TelemetryRequestBlock *b, const uint8_t data_s
     b->bytes[3] |= used_4 && 0x01;
 }
 
+/**
+ * Initializes a GNSS location data block with the provided information
+ * @param b The GNSS location data block to be initialized
+ * @param fix_time The mission time the fix was recieved
+ * @param latitude The latitude of the rocket in units of 100 micro-arcminutes per LSB
+ * @param longitude The longitude of the rocket in units of 100 micro-arcminutes per LSB
+ * @param utc_time The UTC time when the fix was recieved
+ * @param altitude The altitude calculated by the GNSS module, in mm
+ * @param speed The speed over the ground in hundredths of a knot
+ * @param course The course over ground of the rocket in hundredths of a degree
+ * @param pdop Position dilution of precision multiplied by 100
+ * @param hdop Horizontal dilution of precision multipled by 100
+ * @param vdop Vertical dilution of precision multiplied by 100
+ * @param sats The number of GNSS satellites used in the fix
+ * @param fix The fix type, as two bits
+ */
+void gnss_location_db_init(GNSSLocationDB *b, const uint32_t fix_time, const int32_t latitude, const int32_t longitude,
+                           const uint32_t utc_time, const int32_t altitude, int16_t speed, int16_t course,
+                           uint16_t pdop, uint16_t hdop, uint16_t vdop, uint8_t sats, uint8_t fix) {
+    // Could write this all out with sizeofs, but it gets really hard to read after a second
+    memcpy(b->bytes, &fix_time, sizeof(fix_time));
+    memcpy(b->bytes + (4), &latitude, sizeof(latitude));
```

Comment:
Yeah, I see what you mean. I liked `sizeof()` because we could change a parameter later, but I think that's a non-issue and I agree that there would be wayyyy too many `sizeof()` calls here. Looks good!